src/main/java/Filters/DashboardFilter.java

```diff
@@ -0,0 +1,61 @@
+package Filters;
+
+import javax.servlet.*;
+import javax.servlet.annotation.WebFilter;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import java.io.IOException;
+import java.sql.SQLException;
+import java.util.List;
+
+import Models.Event.Event;
+import Models.Event.EventDao;
+import Models.Event.Log;
+import Models.User.*;
+import Utils.Tables;
+
+@WebFilter(filterName = "DashboardFilter")
+public class DashboardFilter implements Filter {
+    @Override
+    public void destroy() {}
+    @Override
+    public void init(FilterConfig config) throws ServletException {}
+
+    @Override
+    public void doFilter(ServletRequest req, ServletResponse resp, FilterChain chain) throws ServletException, IOException {
+        HttpServletRequest request = (HttpServletRequest) req;
+        HttpServletResponse response = (HttpServletResponse) resp;
+        HttpSession session = request.getSession(false);
+
+        User user = (User) session.getAttribute("currentUser");
+
+        switch (user.getRole()) {
+            case "admin":
+                adminDispatcher(request).forward(request, response);
+                break;
+            case "patient":
+                request.getRequestDispatcher("/dashboards/patient.jsp").forward(request, response);
+                break;
+            case "staff":
+                request.getRequestDispatcher("/dashboards/staff.jsp").forward(request, response);
+                break;
+        }
+    }
+
+    private RequestDispatcher adminDispatcher(HttpServletRequest request) {
+        try {
+            if (request.getMethod().equals("POST")) {
+                Tables.recreateTables();
+            }
```

Comment:
Could you talk me through the logic of this if statement, please? Why is it that we would need to recreate tables when redirecting to admin dashboard?

Reply:
I presume you are familiar with how browsers send HTTP methods? If not - when you navigate to a URL, the browser sends a GET request which usually returns the page (though you can be evil and do other stuff). A normal website with javascript would use something like Ajax to send a POST request to a URL endpoint - so a POST request to something like example.com/api/recreateTables. Here, I didn't want a servlet specifically sitting on a route purely dedicated to recreating tables, so instead, if a post request is sent to the admin page we use that as a sign to trigger the recreation of the tables. If we wanted multiple POST functionalities sitting there, we could add the specific functionality we want to trigger as a parameter or on the body of the POST request and expand the if to include a match on that data.

Reply:
It just looks and feels weird because we can only use forms to make POST requests, even when we don't want to load into a different page, just launch some desired functionality. The page reloading is why websites don't really do this anymore, as it leads to weird stuff like pressing a button, the whole page reloading + the functionality of the button happening, then pressing refresh and it 'pressing' the button again.

Reply:
Thanks for explaining mate. I'm guessing the parameter method is technically the "correct" way to go about this, even for a single functionality? Either way, I'm happy to continue with what we have here for time's sake!

Reply:
It would be more obvious what was going on if it had an entire servlet behind it, but we'd end up with a lot of servlets that do very little. In the future if it was expanded it would likely be moved into a Servlet and a POST would match parameters for functionality